Swift/Sources/StateMachine/StateMachine.swift

```diff
@@ -277,10 +312,15 @@ public enum StateMachineTypes {
         }
     }
 
+    fileprivate enum EventType {
+        case normal, onEnter(() -> Void), onExit(() -> Void)
+    }
+
     public struct EventHandler<State: StateMachineHashable, Event: StateMachineHashable, SideEffect> {
 
-        fileprivate let event: Event.HashableIdentifier
-        fileprivate let action: Action<State, Event, SideEffect>.Factory
+        fileprivate let event: Event.HashableIdentifier?
```

Comment:
`?` is probably a wrong approach here.

```
enum eventType {
  normal(event, action),
  onEnter(() -> Void),
  onExit(() -> Void),
}
```
seems more logical type representation.